### ISSUE TITLE
MoneyField in Orchard

### ISSUE THREAD

**Issue by barthamark** [OrchardCMS/Orchard#5989]
There is a Money data type for .NET available open source here: http://www.codeproject.com/Articles/28244/A-Money-type-for-the-CLR
We are using it in several project where money-related operations and data storage is required and it's perfect.

My idea is that we could make a field for that in Orchard called MoneyField.

Note, that it also can manage currencies so there could be a field setting for the currency or just simply a second editor displayed for that (enum) next to the editor of the actual value (decimal).

**Comment by Piedone** [comment]
Orchard just gets too materialistic. I'd rather want to see a LoveField or FamilyField first.

**Comment by barthamark** [comment]
Unfortunately, MoneyField is dependency of LoveField and FamilyField. They are both extremely using Money.dll.

**Comment by Piedone** [comment]
Case in point. :-D

**Comment by sebastienros** [comment]
Like we decided for the other fields that have been suggested, this can be done as a separate module as it's not something most sites would use.

**Comment by sebastienros** [comment]
Other examples: phone, address

**Comment by AlbertoLeon** [comment]
+1 for MoneyField I think if Orchard CMS can be used to show prices in different currencies, is a good solution. The same for Phone or Address. I think have sufficient "business logic" to be independent.

**Comment by bleroy** [comment]
-10,000, and  own an e-commerce module. Which is, coincidentally, where this would belong ;)

**Comment by AlbertoLeon** [comment]
You can need a moneyfield outside e-commerce module... Think in a jobs module that needs to publish the salary to the job position

**Comment by bleroy** [comment]
OK then, create a moneyfiled module, and have both depend on it. Solved!

**Comment by barthamark** [comment]
I agree with Alberto, why would we use an e-commerce module only for displaying a MoneyField in a site that has no connection with any e-commerce operations at all?
Yep, I understand that if it is not in Orchard.Fields module (probably we will see it in a future version) then MoneyField need to live in a relevant module maybe in an open source module.

That's why we started to refactor a few independent and useful fields to a separated module that is going to be published soon.

**Comment by bleroy** [comment]
That's not what I was saying. Yes, the separate module is the way to go. Cool.

**Comment by Mollenhorst** [comment]
Hi Barthamark, did you release the localized 'special fields' module yet? I am curious.

**Comment by Piedone** [comment]
@Mollenhorst Money Fields is included in this module: https://github.com/Lombiq/Lombiq-Fields
